pkg/cmd/target.go

```diff
@@ -996,6 +997,62 @@ func seedWrapper(targetReader TargetReader, targetWriter TargetWriter, configRea
 	return nil
 }
 
+//getAccessRestrictionsFromGardenConfig returns current accessRestrictions from garden config with given garden name
+func getAccessRestrictionsFromGardenConfig(reader ConfigReader, gardenName string) []AccessRestriction {
+	var ars = []AccessRestriction{}
+	config := reader.ReadConfig(pathGardenConfig)
+	for _, garden := range config.GardenClusters {
+		if garden.Name == gardenName && len(garden.AccessRestrictions) > 0 {
+			ars = garden.AccessRestrictions
+		}
+	}
+	return ars
+}
+
+//checkShootsRestriction returns warning message based on comparion between garden config and shoot lables/annotation
+func checkShootsRestriction(shoot gardencorev1beta1.Shoot, reader ConfigReader, gardenName string) string {
+	warningMsg := ""
+	var shootMatchLabels map[string]string
+	var shootAnnotations map[string]string
+	ars := getAccessRestrictionsFromGardenConfig(reader, gardenName)
+	if shoot.Spec.SeedSelector == nil || shoot.Spec.SeedSelector.MatchLabels == nil {
+		return warningMsg
+	}
+	shootMatchLabels = shoot.Spec.SeedSelector.MatchLabels
+	shootAnnotations = shoot.GetAnnotations()
+
+	if len(ars) == 0 {
+		return warningMsg
+	}
+
+	for j := 0; j < len(ars); j++ {
+		ar := ars[j]
+		if _, ok := shootMatchLabels[ar.Key]; ok {
+			if shootMatchLabels[ar.Key] == strconv.FormatBool(ar.NotifyIf) {
+				warningMsg += ar.Msg
+				warningMsg += "\n"
+				//if upper level msg will not show, neither will lower level msg show
+				if len(ar.Options) > 0 {
+					for i := 0; i < len(ar.Options); i++ {
+						option := ar.Options[i]
+						if _, ok := shootAnnotations[option.Key]; ok {
+							if shootAnnotations[option.Key] == strconv.FormatBool(option.NotifyIf) {
+								warningMsg += option.Msg
+								warningMsg += "\n"
+							}
+						}
+
+					}
+				}
+
+			}
+		}
```

Comment:
to simplify nested if statements, use early return pattern. This improves readability